src/networkProviders/proxyNetworkProvider.ts

```diff
@@ -70,77 +80,50 @@ export class ProxyNetworkProvider implements INetworkProvider {
-        );
-
-        // TODO: Fix sorting
-        tokens.sort((a, b) => a.identifier.localeCompare(b.identifier));
-        return tokens;
+    async getAccountStorage(address: Address): Promise<AccountStorage> {
+        const response = await this.doGetGeneric(`address/${address.toBech32()}/keys`);
+        const account = AccountStorage.fromHttpResponse(response);
+        return account;
     }
 
-    async getFungibleTokenOfAccount(
-        address: Address,
-        tokenIdentifier: string,
-    ): Promise<FungibleTokenOfAccountOnNetwork> {
-        const response = await this.doGetGeneric(`address/${address.toBech32()}/esdt/${tokenIdentifier}`);
-        const tokenData = FungibleTokenOfAccountOnNetwork.fromHttpResponse(response.tokenData);
-        return tokenData;
+    async getAccountStorageEntry(address: Address, entryKey: string): Promise<AccountStorageEntry> {
+        const keyAsHex = Buffer.from(entryKey).toString("hex");
+        const response = await this.doGetGeneric(`address/${address.toBech32()}/key/${keyAsHex}`);
+        const account = AccountStorageEntry.fromHttpResponse(response, entryKey);
+        return account;
     }
 
-    async getNonFungibleTokenOfAccount(
-        address: Address,
-        collection: string,
-        nonce: number,
-    ): Promise<NonFungibleTokenOfAccountOnNetwork> {
-        const response = await this.doGetGeneric(
-            `address/${address.toBech32()}/nft/${collection}/nonce/${nonce.valueOf()}`,
-        );
-        const tokenData = NonFungibleTokenOfAccountOnNetwork.fromProxyHttpResponseByNonce(response.tokenData);
-        return tokenData;
+    awaitAccountOnCondition(
+        _address: Address,
+        _condition: (account: AccountOnNetwork) => boolean,
+        options?: AwaitingOptions,
+    ): AccountOnNetwork {
+        if (!options) {
+            options = new AwaitingOptions();
+        }
+        throw new Error("Method not implemented.");
     }
 
-    async getTransaction(txHash: string, _?: boolean): Promise<TransactionOnNetwork> {
-        const url = this.buildUrlWithQueryParameters(`transaction/${txHash}`, { withResults: "true" });
-        const [data, status] = await Promise.all([this.doGetGeneric(url), this.getTransactionStatus(txHash)]);
-        return TransactionOnNetwork.fromProxyHttpResponse(txHash, data.transaction, status);
+    async sendTransaction(tx: Transaction): Promise<string> {
+        const transaction = prepareTransactionForBroadcasting(tx);
+        const response = await this.doPostGeneric("transaction/send", transaction);
+        return response.txHash;
     }
 
-    async getTransactionStatus(txHash: string): Promise<TransactionStatus> {
-        const response = await this.doGetGeneric(`transaction/${txHash}/process-status`);
-        const status = new TransactionStatus(response.status);
-        return status;
+    async simulateTransaction(tx: Transaction, checkSignature: boolean = false): Promise<any> {
```

Comment:
just as an ideea, maybe we should set a dummy signature if not set. Maybe it'll improve dev experience.